common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/OfflineEntitlementsManager.kt

```diff
@@ -1,18 +1,100 @@
 package com.revenuecat.purchases.common.offlineentitlements
 
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.PurchasesErrorCode
+import com.revenuecat.purchases.common.AppConfig
 import com.revenuecat.purchases.common.Backend
 import com.revenuecat.purchases.common.caching.DeviceCache
 import com.revenuecat.purchases.common.debugLog
 import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.common.warnLog
 import com.revenuecat.purchases.strings.OfflineEntitlementsStrings
 
 class OfflineEntitlementsManager(
+    private val appConfig: AppConfig,
     private val backend: Backend,
+    private val offlineCustomerInfoCalculator: OfflineCustomerInfoCalculator,
     private val deviceCache: DeviceCache
 ) {
+    // We cache the offline customer info in memory, so it's not persisted.
+    val offlineCustomerInfo: CustomerInfo?
+        get() = _offlineCustomerInfo
+    @get:Synchronized @set:Synchronized
+    private var _offlineCustomerInfo: CustomerInfo? = null
+
+    private val offlineCustomerInfoCallbackCache = mutableMapOf<String, List<OfflineCustomerInfoCallback>>()
+
+    fun resetOfflineCustomerInfoCache() {
+        synchronized(this) {
+            if (_offlineCustomerInfo != null) {
```

Comment:
Well, I wanted to log a message here, when changing from not null to null, since that's basically "deactivating" offline entitlements mode. So yeah, it's only for a log but I think it could be useful, so would prefer to leave it... Let me know if you feel strongly about it though!

Reply:
Makes sense. I knew there was a reason for it! Leave it as is then